ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -1123,6 +1123,11 @@ func getTaskMetadataErrorResponse(endpointContainerID, requestType string, err e
 		return http.StatusInternalServerError, fmt.Errorf("unable to obtain container metadata for container: %s", endpointContainerID)
 	}
 
+	var errDefaultNetworkInterfaceName *state.ErrorDefaultNetworkInterfaceName
+	if errors.As(err, &errDefaultNetworkInterfaceName) {
+		return http.StatusInternalServerError, fmt.Errorf("unable to obtain default network interface name on host for container: %s", endpointContainerID)
```

Comment:
We don't include the `externalReason` in the response. Is that expected?

Reply:
hmm we can but I was thinking we shouldn't return the internal/specific error back as the response. will be logging it instead

Reply:
response body will now include externalReason. thanks!